index.js

```diff
@@ -1,16 +1,37 @@
 'use strict';
 
+class PriorityQueue {
+	constructor() {
+		this._queue = [];
+	}
+	_compare(a, b) {
+		return b.priority - a.priority;
+	}
+	push(run, priority) {
+		priority = priority || 0;
+		this._queue.push({priority, run});
+		this._queue.sort(this._compare);
+	}
+	shift() {
+		return this._queue.shift().run;
+	}
+	get length() {
+		return this._queue.length;
+	}
```

Comment:
I've never been a fan of the array method naming, especially `shift`, which I always forget the name of.

I would rather name these: `add`, `remove`, `size`.